tests/aws-cpp-sdk-core-tests/smithy/client/SmithyClientTest.cpp

```diff
@@ -19,16 +27,162 @@ class SmithyClientTest : public Aws::Testing::AwsCppSdkGTestSuite {
 class MyTestEndpointProvider : public Aws::Endpoint::EndpointProviderBase<>
 {
 };
-
+using MySmithyClientConfig = Aws::Client::ClientConfiguration;
+using MyServiceAuthSchemeResolver = smithy::AuthSchemeResolverBase<smithy::DefaultAuthSchemeResolverParameters>; //smithy::SigV4AuthSchemeResolver<>; 
 static constexpr char MyServiceName[] = "MySuperService";
+using SigVariant = Aws::Crt::Variant<smithy::SigV4AuthScheme, smithy::SigV4aAuthScheme>;
+using MySmithyClient = smithy::client::AwsSmithyClientT<MyServiceName,
+                                                        MySmithyClientConfig,
+                                                        MyServiceAuthSchemeResolver,
+                                                        SigVariant,
+                                                        MyTestEndpointProvider>;
+
+
+class TestClient : public MySmithyClient
+{
+    public:
+    TestClient(const MySmithyClientConfig& clientConfig,
+            const Aws::String& serviceName,
+            const std::shared_ptr<Aws::Http::HttpClient>& httpClient,
+            const std::shared_ptr<Aws::Client::AWSErrorMarshaller>& errorMarshaller,
+            const std::shared_ptr<MyTestEndpointProvider> endpointProvider,
+            const std::shared_ptr<MyServiceAuthSchemeResolver>& authSchemeResolver,
+            const Aws::UnorderedMap<Aws::String, SigVariant>& authSchemesMap):
+            MySmithyClient(
+                clientConfig,
+                serviceName,
+                httpClient,
+                errorMarshaller,
+                endpointProvider,
+                authSchemeResolver,
+                authSchemesMap)
+    {}
+
+
+    SelectAuthSchemeOptionOutcome SelectAuthSchemeOption(const smithy::client::AwsSmithyClientAsyncRequestContext& ctx) const
+    {
+        return MySmithyClient::SelectAuthSchemeOption(ctx);
+    }
+
+    SigningOutcome SignRequest(std::shared_ptr<HttpRequest> httpRequest, const smithy::AuthSchemeOption& targetAuthSchemeOption) const
+    {
+        return MySmithyClient::SignRequest(httpRequest, targetAuthSchemeOption);
+    }
+
+
+};
+
+
+
+TEST_F(SmithyClientTest, testSigV4) {
+
+    const char ALLOCATION_TAG[] = "SmithyClientTest";
+
+    Aws::Client::ClientConfiguration clientConfig;
+    std::shared_ptr<Aws::Http::HttpClient> httpClient = Aws::Http::CreateHttpClient(clientConfig);
```

Comment:
should we be creating a real http client or using the [mock http client](MockHttpClient)? we likely dont want to be creating a real http client for unit tests.

Reply:
This is created for integration test to see where it actually fails in the workflow. Some parts are yet to be validated. I agree for unit tests we should use mock

Reply:
I would like this to be an integration test to create an actual signature as i want to test the auth using the aws credentials api calls

Reply:
an HTTP client is not required for a signature, only to send a request, testing signing in this context would be a unit tests, when you actually call a aws service it would be a integration test